sdl_android/src/main/java/com/smartdevicelink/transport/MultiplexBluetoothTransport.java

```diff
@@ -257,9 +254,9 @@ public synchronized void connected(BluetoothSocket socket, BluetoothDevice devic
         	mConnectedWriteThread = null;
         }
         // Cancel the accept thread because we only want to connect to one device
-        if (!keepSocketAlive && getBluetoothSerialServerInstance().mSecureAcceptThread != null) {
-        	getBluetoothSerialServerInstance().mSecureAcceptThread.cancel();
-            getBluetoothSerialServerInstance().mSecureAcceptThread = null;
+        if (!keepSocketAlive && mSecureAcceptThread != null) {
+        	mSecureAcceptThread.cancel();
+            mSecureAcceptThread = null;
         }
 
         // Start the thread to manage the connection and perform transmissions
```

Comment:
`if(device!=null && device.getName()!=null && device.getName()!="")` this statement behaves the way we expect it to be because the empty string is interned. But it almost always should be: `(.... && device.getName().equals("")`. 
Actually the whole statement can just be: `if (!("").equals(device.getName())`